clients/admin-ui/src/features/configure-consent/AddVendor.tsx

```diff
@@ -101,19 +120,48 @@ const AddVendor = ({
     helpers: FormikHelpers<FormValues>
   ) => {
     const transformedDeclarations = values.privacy_declarations
-      .filter((dec) => dec.data_use !== EMPTY_DECLARATION.data_use)
-      .map((dec) => {
+      .filter((dec) => dec.consent_use !== EMPTY_DECLARATION.consent_use)
+      .flatMap((dec) => {
         // if a cookie from the form already exists on the declaration with full
         // information from the dictionary, use that; otherwise, make the cookie
         // name from the form into a new cookie
         const transformedCookies = dec.cookieNames.map((name) => {
           const existingCookie = dec.cookies.find((c) => c.name === name);
           return existingCookie ?? { name, path: "/" };
         });
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        const { cookieNames, consent_use, ...rest } = dec;
 
-        const { cookieNames, ...rest } = dec;
-        return { ...rest, cookies: transformedCookies };
+        // for "marketing", we create two data uses on the backend
+        if (dec.consent_use === "marketing" && !dec.data_use) {
+          return [
+            "marketing.advertising.first_party.targeted",
+            "marketing.advertising.third_party.targeted",
+          ].map((dataUse) => ({
+            ...rest,
+            data_use: dataUse,
+            cookies: transformedCookies,
+          }));
+        }
+        return {
+          ...rest,
+          data_use: dec.data_use ? dec.data_use : dec.consent_use!,
+          cookies: transformedCookies,
+        };
       });
+    // if editing and the system has existing data uses not shown on form
+    // due to not being consent uses, include those in the payload
+    const existingDeclarations = passedInSystem
+      ? passedInSystem.privacy_declarations.filter(
+          (du) =>
+            !consentUseOptions.some(
+              (opt) => opt.value === du.data_use.split(".")[0]
```

Comment:
we're using this logic quite a bit, might be worth splitting out into a simple function? i.e. `dataUseIsAConsentUse(du)`